### PR TITLE
:sparkles: Scaffold auth

### PR DESCRIPTION
**Pull request by mengqiy**
The 1st commit update CR and CT to pickup latest changes that is necessary for this PR.

### CODE REVIEW

File: pkg/scaffold/manager/cmd.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ import (
 )
 
 func main() {
+	var metricsBindAddr string
+	flag.StringVar(&metricsBindAddr, "metrics-bind-addr", ":8080", "The address the metric endpoint binds to.")
```
Comment (droot) [inline pkg/scaffold/manager/cmd.go]: nit: can we call the flag `metrics-addr` ? (internally we can use metricsBindAddr)

File: pkg/scaffold/project/kustomize_proxy_patch.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"path/filepath"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/input"
+)
+
+var _ input.File = &KustomizeProxyPatch{}
+
+// KustomizeProxyPatch scaffolds the patch file for enabling
+// prometheus metrics for manager Pod.
+type KustomizeProxyPatch struct {
```
Comment (droot) [inline pkg/scaffold/project/kustomize_proxy_patch.go]: Can we call this AuthProxy instead of Proxy ?

File: pkg/scaffold/project/kustomize.go [hunk 3]
```diff
@@ -74,10 +74,26 @@ resources:
 - ../rbac/rbac_role.yaml
 - ../rbac/rbac_role_binding.yaml
 - ../manager/manager.yaml
+# Comment the following 3 lines if you want to disable
+# kube-rbac-proxy (https://github.com/brancz/kube-rbac-proxy)
+# which protects your /metrics endpoint.
+- ../rbac/kube-rbac-proxy-service.yaml
```
Comment (droot) [inline pkg/scaffold/project/kustomize.go]: Can we name this file `auth_proxy_service.yaml` to keep it consistent with the roles file (also note the underscore ?

File: pkg/scaffold/project/kustomize.go [hunk 4]
```diff
@@ -74,10 +74,26 @@ resources:
...
+- ../rbac/auth_proxy_role_binding.yaml
 
 patches:
 - manager_image_patch.yaml
-- manager_prometheus_metrics_patch.yaml
+
+# Protect the /metrics endpoint by putting it behind auth.
+# Only one of manager_kube_rbac_proxy_patch.yaml and
+# manager_prometheus_metrics_patch.yaml should be enabled.
+- manager_kube_rbac_proxy_patch.yaml
```
Comment (droot) [inline pkg/scaffold/project/kustomize.go]: Can we rename this file to use manager_auth_proxy_patch.yaml ?

File: pkg/scaffold/project/kustomize.go [hunk 5]
```diff
@@ -74,10 +74,26 @@ resources:
...
+# If you want your controller-manager to expose the /metrics
+# endpoint w/o any authorization, uncomment the following line and
```
Comment (droot) [inline pkg/scaffold/project/kustomize.go]: w/o any authn/authz ?

**Comment by mengqiy** [comment]
@droot The last 2 commits are incremental changes since your last review.

File: Gopkg.toml [hunk 6]
```diff
@@ -5,6 +5,7 @@ required = [
     "github.com/go-openapi/spec",
     "github.com/onsi/ginkgo", # for integration testing
     "github.com/spf13/pflag",
+    "github.com/pkg/errors",
```
Comment (mengqiy) [inline Gopkg.toml]: dep will drop it w/o this line even it's a dependency of sigs.k8s.io/controller-tools

**Comment by droot** [comment]
Looks good to me. Pl. squash the commits and then we are good to do.

**Comment by mengqiy** [comment]
> Looks good to me. Pl. squash the commits and then we are good to do.

Done

**Pull request merged**